tasks/notify_error.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from dataclasses import dataclass
+
+from asgiref.sync import async_to_sync
+from sqlalchemy.orm import Session
+
+from celery_config import notify_error_task_name
+from database.enums import ReportType
+from database.models import Commit, CommitReport, Upload
+from helpers.checkpoint_logger import from_kwargs as checkpoints_from_kwargs
+from helpers.checkpoint_logger.flows import UploadFlow
+from helpers.notifier import BaseNotifier, NotifierResult
+from services.yaml import UserYaml
+from tasks.base import BaseCodecovTask, celery_app
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class ErrorNotifier(BaseNotifier):
+    failed_upload: int
+    total_upload: int
+
+    def build_message(
+        self,
+    ) -> str:
+        error_message = f"❗️ We couldn't process [{self.failed_upload}] out of [{self.total_upload}] uploads. Codecov cannot generate a coverage report with partially processed data. Please review the upload errors on the commit page."
+        return error_message
+
+
+class NotifyErrorTask(BaseCodecovTask, name=notify_error_task_name):
+    def run_impl(
+        self,
+        db_session: Session,
+        *,
+        repoid: int,
+        commitid: str,
+        current_yaml=None,
+        **kwargs,
+    ):
+        log.info(
+            "Starting notify error task",
+            extra=dict(commit=commitid, repoid=repoid, commit_yaml=current_yaml),
+        )
+
+        # get all upload errors for this commit
+        #
+        commit_yaml = UserYaml.from_dict(current_yaml)
+
+        checkpoints = checkpoints_from_kwargs(UploadFlow, kwargs)
+
+        commits_query = db_session.query(Commit).filter(  # type:ignore
+            Commit.repoid == repoid,
+            Commit.commitid == commitid,  # type:ignore
+        )
+
+        commit: Commit = commits_query.first()
+        assert commit, "Commit not found in database."
```

Comment:
can you explain this line? are you making sure that the `commit` does not exist? why does the `.first()` function return as that message?

Reply:
it's the opposite, we're making sure that commit exists, and if it doesn't we throw the assertion error with that message

Reply:
aaahhhh, i was reading it as though the `assert` was asserting that `element 1` and `element 2` were equal, rather than that `element 1` is truthy else throw `element 2`